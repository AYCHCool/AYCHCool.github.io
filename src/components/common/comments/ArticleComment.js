import React from 'react';
import './articleComment.scss';

const ArticleComment = () => (
  <div className="article__comments">
    <div className="article__comment_card">
      <div className="user__comment_info">
        <img
          src="https://image.flaticon.com/icons/svg/147/147144.svg"
          alt="article author"
          width="50"
          className="circle"
        />
        <div className="comment_details">
          <p className="commenter__name">Jane Doe</p>
          <p className="commenter__time">3 Jun 2019</p>
        </div>
      </div>
      <div className="comment__body">
        <p>
                I don’t know how people don’t realize their employers are tracking what they do at
                this point. I never keep my work email on my phone and deleted slack as well.
                If anyone needs me after work hours they can call me!
        </p>
      </div>
      <div className="comment__likes">
        <i id="comment__likes__icon" className="material-icons">favorite_border</i>
        <p className="comment__likes__count">33</p>
      </div>
    </div>
    <div className="article__comment_card">
      <div className="user__comment_info">
        <img
          src="https://image.flaticon.com/icons/svg/147/147144.svg"
          alt="article author"
          width="50"
          className="circle"
        />
        <div className="comment_details">
          <p className="commenter__name">Obiwan Kenobi</p>
          <p className="commenter__time">4 Jun 2019</p>
        </div>
      </div>
      <div className="comment__body">
        <p>
                I don’t know how people don’t realize their employers are tracking what they do at
                this point. I never keep my work email on my phone and deleted slack as well.
                If anyone needs me after work hours they can call me!
        </p>
      </div>
      <div className="comment__likes">
        <i id="comment__likes__icon" className="material-icons">favorite_border</i>
        <p className="comment__likes__count">12</p>
      </div>
    </div>
    <div />
  </div>
);

export default ArticleComment;
