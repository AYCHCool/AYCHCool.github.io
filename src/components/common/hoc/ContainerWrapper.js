/* eslint-disable react/prop-types */
import React from 'react';
import { Route } from 'react-router-dom';
import Header from '@Common/header';

const ContainerWrapper = ({ component: Component, ...rest }) => (
  <Route
    {...rest}
    render={props => (
      <>
        <Header {...props} />
        <div className="container">
          <Component {...props} />
        </div>
      </>
    )}
  />
);


export default ContainerWrapper;
