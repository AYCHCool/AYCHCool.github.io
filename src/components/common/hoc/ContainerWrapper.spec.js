import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import ContainerWrapper from './ContainerWrapper';
import ArticleComment from '../comments/ArticleComment';

const initialState = {
  auth: {
    isAuthenticated: false,
    user: {},
  },
};

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore(initialState);

describe('Container wrapper higher order component', () => {
  const props = {
    component: () => <ArticleComment />,
    render: () => ({}),
  };

  it('should render the component without crashing', () => {
    const wrapper = shallow(<ContainerWrapper {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render and return another component wrapped in a container', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <ContainerWrapper {...props} />
        </Router>
      </Provider>,
    );
    expect(wrapper.find('.container').length).toEqual(1);
    expect(wrapper.find('Route').length).toEqual(1);
    wrapper.unmount();
  });
});
