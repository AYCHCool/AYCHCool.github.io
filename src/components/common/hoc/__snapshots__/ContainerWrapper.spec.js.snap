// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container wrapper higher order component should render the component without crashing 1`] = `ShallowWrapper {}`;
